yarn-project/aztec.js/src/wallet/base_wallet.ts

```diff
@@ -94,90 +94,50 @@ export abstract class BaseWallet implements Wallet {
     return account.createAuthWit(messageHashOrIntent);
   }
 
-  // docs:start:estimateGas
   /**
-   * Estimates gas for a given tx request and returns gas limits for it.
-   * @param executionPayload - The execution payload to calculate the gas for
-   * @param opts - Options.
-   * @returns Gas limits.
+   * Returns default values for the transaction fee options
+   * if they were omitted by the user.
+   * @param address - The account building the transaction, usually the one paying the fee
+   * @param userFeeOptions - User-provided fee options, which might be incomplete
```

Comment:
```suggestion
   * User-provided fee options provided on the input contain the same values as the return value type with
   * the difference that the values in UserFeeOptions are optional. This function populates the missing values
   * in UserFeeOptions with defaults.
   *
   * @param address - The account building the transaction, usually the one paying the fee
   * @param userFeeOptions - User-provided fee options to use as a basis for the output
```
Before I needed to go check what exactly are UserFeeOptions to understand what this does. Think with this change it's now unnecessary.

Please copy this to other implementations of this func if you agree with it.

Reply:
Maybe we could even rename `UserFeeOptions` to something like`PartialFeeOptions`. Then just the types of inputs and outputs of the function would make it quite clear what it does. WDYT?

Reply:
This type is being completely reworked into `FeeOptionsHint` in the next PR